### PR TITLE
Fix adding colors to swatches not updating in previous ColorPickers.

### PR DESCRIPTION
**Pull request by anvilfolk**
ColorPicker was only updating colors if its swatches were empty. It should always update from the cache in case some other ColorPicker updated the swatch cache.

You can see the error within ~30 seconds in this [stream](https://www.youtube.com/watch?v=s0dmBDtkH3g&t=5422s).

### CODE REVIEW

File: scene/gui/color_picker.cpp [hunk 1]
```diff
@@ -608,13 +608,14 @@ void ColorPicker::_update_presets() {
 
 #ifdef TOOLS_ENABLED
 	if (editor_settings) {
-		// Only load preset buttons when the only child is the add-preset button.
-		if (preset_container->get_child_count() == 1) {
-			for (int i = 0; i < preset_cache.size(); i++) {
-				_add_preset_button(preset_size, preset_cache[i]);
-			}
-			_notification(NOTIFICATION_VISIBILITY_CHANGED);
+		// Always update according to cache
```
Comment (KoBeWi) [inline scene/gui/color_picker.cpp:611]:
```suggestion
		// Always update according to cache.
```

File: scene/gui/color_picker.cpp [hunk 2]
```diff
@@ -608,13 +608,14 @@ void ColorPicker::_update_presets() {
...
+		for (int i = 1; i < preset_container->get_child_count(); i++) {
+			Object::cast_to<ColorPresetButton>(preset_container->get_child(i))->queue_free();
```
Comment (KoBeWi) [inline scene/gui/color_picker.cpp:613]:
```suggestion
			preset_container->get_child(i)->queue_free();
```
No need to cast.

File: scene/gui/color_picker.cpp (hunk 1 above)
Comment (KoBeWi) [reply scene/gui/color_picker.cpp:611]: Although the comment is not really necessary now. You could remove it.

**Review by KoBeWi — APPROVED** [review]

File: scene/gui/color_picker.cpp (hunk 2 above)
Comment (anvilfolk) [reply scene/gui/color_picker.cpp]: Good point!

File: scene/gui/color_picker.cpp (hunk 1 above)
Comment (anvilfolk) [reply scene/gui/color_picker.cpp]: You're right, I thought about removing then ended up opting to use this opportunity to clarify why the first button isn't removed as well :)

**Comment by anvilfolk** [comment]
Changes made and rebased, thanks for reviewing! :)

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.4.

**Pull request merged**